sdk/dotnet/Extensions.cs

```diff
@@ -0,0 +1,64 @@
+﻿// Copyright 2016-2019, Pulumi Corporation.
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+
+namespace Pulumi.Docker
+{
+    internal static class UnwrapExtensions
+    {
+        public static Input<ImmutableArray<T>?> Unwrap<T>(this InputList<T>? items)
+        {
+            return items == null ? Output.Create((ImmutableArray<T>?)null) : items.Apply(v => (ImmutableArray<T>?)v);
+        }
+
+        public static Input<T?> Unwrap<T>(this Input<T>? input) where T : class
+        {
+            return input == null ? Output.Create((T?)null) : input.Apply(v => (T?)v);
+        }
```

Comment:
these seems like useful enough to move to the Pulumi layer.  can you file issue/PR about that?

Reply:
I split the class to `Extensions` (generic) and `DockerExtension` (specific). There are a log of potential generic API based on combinations of inputs, lists, maps, unions, nullable/non-nullable, class/struct, plain/output/input, etc. So, I'm torn on the value of my extensions: are they useful enough to warrant the complexity? What are good names for them?
Looking forward to your opinion.